O365/utils/token.py

```diff
@@ -44,6 +44,26 @@ def expiration_datetime(self):
             expires_on = expires_on + dt.timedelta(days=90)
         return expires_on
 
+    @property
+    def access_expiry(self):
+        """
+        Returns the token's access expiration datetime
+        :return datetime: The datetime the token's access expires
+        """
+        expires_at = self.get('expires_at')
+        if expires_at:
+            return dt.datetime.fromtimestamp(expires_at)
```

Comment:
The `expiration_datetime` can now get the expiration value from this method and then add the 90 days if the token is long lived. This way we don't repeat code.

So you should substract the EXPIRES_ON_THRESHOLD. I think two minutes of threshold is maybe too much,.

Reply:
agreed...removed repeat code & updated threshold to 1min

curious - what is the reason you have this threshold?

Reply:
Because if someone is checking the `is_expired` property with a refresh token enabled I want the user to be able to request the new refresh token before it expires. It just an extremely edge case, but I thought 2 minutes in a 90 day range is nothing.